recipes/boost/all/conanfile.py

```diff
@@ -368,14 +222,18 @@ def _all_super_modules(self, name):
                 break
         return dependencies
 
+    def _disable_component(self, name):
+        super_modules = self._all_super_modules(name)
+        for smod in super_modules:
+            try:
+                setattr(self.options, f"without_{smod}", True)
+            except ConanException:
+                pass
```

Comment:
Doesn't this always raise `ConanException`? That is, the modules are not actually disabled by this logic?